autoload/go/config.vim

```diff
@@ -240,20 +240,42 @@ function! go#config#SetTemplateAutocreate(value) abort
   let g:go_template_autocreate = a:value
 endfunction
 
+function! go#config#Metalinter() abort
+  return get(g:, "go_metalinter", "gometalinter")
+endfunction
+
 function! go#config#MetalinterCommand() abort
   return get(g:, "go_metalinter_command", "")
 endfunction
 
 function! go#config#MetalinterAutosaveEnabled() abort
-  return get(g:, 'go_metalinter_autosave_enabled', ['vet', 'golint'])
+  let l:default_enabled = ["vet", "golint"]
+
+  if go#config#Metalinter() == "golangci-lint"
+    let l:default_enabled = ["deadcode", "errcheck", "ineffassign", "structcheck", "typecheck", "varcheck"]
```

Comment:
Why have you chosen these as the defaults?

At a minimum, I'd expect the defaults for golangci-lint to at least use `golint` and `govet` to maintain parity with gometalinter for now.

Reply:
These are the "fast" defaults of golangci-lint https://github.com/golangci/golangci-lint#enabled-by-default-linters. There's also a `--fast` flag available to use, but we need to pass it in as another config because the non-autosave command shouldn't be restricted to "fast" linters.

I personally use `golint`, but it's not part of their defaults, so I didn't throw it in. I would be happy putting it in.

Their `govet` seems a bit odd to be honest, I haven't gotten it to mimic the output of a regular `go vet ./...`. It seems like they do some exclude duplicate checks between linters, so typecheck which is default takes over some of the govet behaviors. And then their `govet` just does some other source code analysis.

Reply:
Yeah, let's put in `golint`.

Reply:
added